bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -110,72 +98,113 @@ public WemoHandler(Thing thing, UpnpIOService upnpIOService, WemoHttpCall wemoHt
-            if (command instanceof OnOffType) {
-                try {
-                    String binaryState = null;
+        }
+    }
 
-                    if (command.equals(OnOffType.ON)) {
-                        binaryState = "1";
-                    } else if (command.equals(OnOffType.OFF)) {
-                        binaryState = "0";
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (host != null && !host.isEmpty()) {
+            String wemoURL = getWemoURL(host, "basicevent");
+            if (wemoURL != null) {
+                if (command instanceof RefreshType) {
+                    try {
+                        updateWemoState();
+                    } catch (Exception e) {
+                        logger.debug("Exception during poll", e);
                     }
-
-                    String soapHeader = "\"urn:Belkin:service:basicevent:1#SetBinaryState\"";
-
-                    String content = "<?xml version=\"1.0\"?>"
-                            + "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\" s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"
-                            + "<s:Body>" + "<u:SetBinaryState xmlns:u=\"urn:Belkin:service:basicevent:1\">"
-                            + "<BinaryState>" + binaryState + "</BinaryState>" + "</u:SetBinaryState>" + "</s:Body>"
-                            + "</s:Envelope>";
-
-                    URL descriptorURL = service.getDescriptorURL(this);
-                    String wemoURL = getWemoURL(descriptorURL, "basicevent");
-
-                    if (wemoURL != null) {
-                        wemoCall.executeCall(wemoURL, soapHeader, content);
+                } else if (channelUID.getId().equals(CHANNEL_STATE)) {
+                    if (command instanceof OnOffType) {
+                        try {
+                            String binaryState = null;
+
+                            if (command.equals(OnOffType.ON)) {
+                                binaryState = "1";
+                            } else if (command.equals(OnOffType.OFF)) {
+                                binaryState = "0";
+                            }
+                            String soapHeader = "\"urn:Belkin:service:basicevent:1#SetBinaryState\"";
+                            String content = "<?xml version=\"1.0\"?>"
+                                    + "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\" s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"
+                                    + "<s:Body>" + "<u:SetBinaryState xmlns:u=\"urn:Belkin:service:basicevent:1\">"
+                                    + "<BinaryState>" + binaryState + "</BinaryState>" + "</u:SetBinaryState>"
+                                    + "</s:Body>" + "</s:Envelope>";
+                            wemoCall.executeCall(wemoURL, soapHeader, content);
```

Comment:
Refactoring suggestion to increase readability: Extract this to private method taking boolean parameter, so it would be called from here like this: `setWemoState(OnOffType.ON == command);`

Reply:
refactored a bit different... Will think about moving the content builder to WemoUtil class....